MergerLogic/DataTypes/S3.cs

```diff
@@ -71,57 +73,61 @@ private List<int> GetZoomLevels()
             return zoomLevels;
         }
 
-        public override List<Tile> GetNextBatch(out string batchIdentifier)
+        public override List<Tile> GetNextBatch(out string currentBatchIdentifier,out string? nextBatchIdentifier, string? inCompletedBatchIdentifier, long? totalTilesCount)
         {
-            batchIdentifier = this._continuationToken;
-            List<Tile> tiles = new List<Tile>();
-            int missingTiles = this.BatchSize;
-
-            while (missingTiles > 0)
+            lock (_locker)
             {
-                if (this._endOfRead && !this._zoomEnumerator.MoveNext())
-                {
-                    break;
-                }
-
-                string path = $"{this.Path}/{this._zoomEnumerator.Current}/";
-
-                var listRequests = new ListObjectsV2Request
+                
+                currentBatchIdentifier = this._continuationToken ?? nullStringValue;
```

Comment:
Was this logic checked? Do we know how to handle this?

Reply:
yes

Reply:
Amazing!